src/hotspot/share/code/icBuffer.hpp

```diff
@@ -62,16 +62,21 @@ class ICStub: public Stub {
   // General info
   int     size() const                           { return _size; }
 
-  // ICStub_from_destination_address looks up Stub* address from code entry address,
-  // which unfortunately means the stub head should be at the same alignment as the code.
-  static  int alignment()                        { return CodeEntryAlignment; }
+  // For extra correctness/safety, we want to make sure that each ICStub is in a separate
+  // instruction cache line. This would rely on instruction cache coherency to order
+  // the updates to ICStub and the setting the destination to the ICStub.
```

Comment:
Maybe it would on x86, but I can't see how that works on AArch64. On AArch64 we don't depend on any of that.
Mark this comment as x86?

Reply:
I added `on some architectures`, hopefully it is enough to break the over-generalization.